app/controllers/uploads_controller.rb

```diff
@@ -321,6 +329,13 @@ def create_multipart
         debug_upload_error(err, "upload.create_mutlipart_failure", additional_detail: err.message),
         status: 422
       )
+    rescue BackupRestore::BackupStore::BackupFileExists
```

Comment:
This is some of the jankyness mentioned in the PR description that I hope to address in a subsequent PR.